controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -211,6 +222,34 @@ public void deleteStream(final String scopeName, final String streamName, final
         .thenAccept(x -> LoggerHelpers.traceLeave(log, "deleteStream", traceId));
     }
 
+    @Override
+    public void getReaderGroup(final String scopeName, final String readerGroupName,
+                               final SecurityContext securityContext, final AsyncResponse asyncResponse) {
+        long traceId = LoggerHelpers.traceEnter(log, "getReaderGroup");
+
+        LocalController controller = new LocalController(controllerService);
+        ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(scopeName, controller,
+                new ClientFactoryImpl(scopeName, controller), new ConnectionFactoryImpl(false));
+        ReaderGroupProperty readerGroupProperty = new ReaderGroupProperty();
+        readerGroupProperty.setScopeName(scopeName);
+        readerGroupProperty.setReaderGroupName(readerGroupName);
+        CompletableFuture.supplyAsync(() -> {
+            ReaderGroup readerGroup = readerGroupManager.getReaderGroup(readerGroupName);
+            readerGroupProperty.setOnlineReaderIds(
+                    new ArrayList<>(readerGroup.getOnlineReaders()));
+            readerGroupProperty.setStreamList(
+                    new ArrayList<>(readerGroup.getStreamNames()));
+            return Response.status(Status.OK).entity(readerGroupProperty).build();
+        }, controllerService.getExecutor()).exceptionally(exception -> {
+            log.warn("getReaderGroup for {} failed with exception: ", readerGroupName, exception);
+            return Response.status(Status.INTERNAL_SERVER_ERROR).build();
```

Comment:
Done. returning 404 now for readergroup or scope not found. Also included this in system and integration tests